OpenRA.Mods.D2k/Traits/Render/WithDecorationCarryable.cs

```diff
@@ -0,0 +1,38 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.D2k.Traits
+{
+	[Desc("Displays a sprite when the carryable actor is waiting for pickup.")]
+	public class WithDecorationCarryableInfo : WithDecorationInfo, Requires<CarryableInfo>
+	{
+		public override object Create(ActorInitializer init) { return new WithDecorationCarryable(init.Self, this); }
+	}
+
+	public class WithDecorationCarryable : WithDecoration
+	{
+		readonly Carryable carryable;
+
+		public WithDecorationCarryable(Actor self, WithDecorationCarryableInfo info)
+			: base(self, info)
+		{
+			carryable = self.Trait<Carryable>();
+		}
+
+		public override bool ShouldRender(Actor self)
+		{
+			return carryable.WantsTransport;
```

Comment:
Because the carryall code is still not as good as it should be and this will add an almost permanent arrow to the harvester.
The point is to know when there is a carryall coming for it. It will almost always want one.